Source/WebInspectorUI/UserInterface/Views/MediaDetailsSidebarPanel.js

```diff
@@ -0,0 +1,240 @@
+    }
+
+    get videoFormat() {
+        return this._videoFormat;
+    }
+
+    set videoFormat(videoFormat) {
+        if (this._videoFormat === videoFormat)
+            return;
+
+        this._videoFormat = videoFormat;
+        this._videoFormatRow.value = videoFormat;
+    }
+
+    get audioFormat() {
+        return this._audioFormat;
+    }
+
+    set audioFormat(audioFormat) {
+        if (this._audioFormat === audioFormat)
+            return;
+
+        this._audioFormat = audioFormat;
+        this._audioFormatRow.value = audioFormat;
+    }
+
+    get video() {
+        return this._video;
+    }
+
+    set video(video) {
+        if (this._video?.width === video?.width
+            && this._video?.height === video?.height
+            && this._video?.framerate === video?.framerate
+            && this._video?.codec === video?.codec
+            && this._video?.colorSpace?.transfer === video?.colorSpace?.transfer
+            && this._video?.colorSpace?.primaries === video?.colorSpace?.primaries
+            && this._video?.colorSpace?.matrix === video?.colorSpace?.matrix
+            && this._video?.fullRange === video?.fullRange)
+            return;
+
+        this._video = video;
+        this._videoSection.element.classList.toggle("hidden", !video);
+        this._resolutionRow.value = WI.UIString("%dx%d (%dfps)").format(video?.width ?? 0, video?.height ?? 0, video?.framerate ?? 0);
+        this._resolutionRow.element.classList.toggle("hidden", !video);
+        this._videoCodecRow.value = video?.codec;
+        this._transferRow.value = video?.colorSpace?.transfer;
+        this._primariesRow.value = video?.colorSpace?.primaries;
+        this._matrixRow.value = video?.colorSpace?.matrix;
+        this._fullRangeRow.value = video?.fullRange ? WI.UIString("Full range") : WI.UIString("Video range");
+    }
+
+    get audio() {
+        return this._audio;
+    }
+
+    set audio(audio) {
+        if (this._audio?.codec === audio?.codec
+            && this._audio?.sampleRate === audio?.sampleRate
+            && this._audio?.numberOfChannels === audio?.numberOfChannels)
+            return;
+
+        this._audio = audio;
+        this._audioSection.element.classList.toggle("hidden", !audio);
+        this._audioCodecRow.value = audio?.codec;
+        this._sampleRateRow.value = WI.UIString("%d Hz").format(audio?.sampleRate);
+        switch (audio?.numberOfChannels) {
+        case 1:
+            this._channelsRow.value = WI.UIString("Mono");
+            break;
+        case 2:
+            this._channelsRow.value = WI.UIString("Stereo");
+            break;
+        default:
+            this._channelsRow.value = audio?.numberOfChannels ?? WI.UIString("Unknown");
+            break;
+        }
+    }
+
+    attached() {
+        super.attached();
+        if (this._updateTimer)
+            clearInterval(this._updateTimer);
+        this._updateTimer = setInterval(() => this.needsLayout(), 250);
```

Comment:
It isn't generally a good idea to poll like this for updates. The backend should be sending an event over the protocol with updates when they exist instead of the frontend having to continually ask the backend. Additionally, a fixed interval doesn't take into account it may take longer than that interval to perform the update, at which point you may have multiple updates in flight at once, or worse you may have a growing backlog of update requests still waiting on the backend.

Reply:
There's no event for the most frequent updated value in these stats (the dropped frame count). It's a poll-or-nothing here.

Reply:
I see. Polling it is, I guess. I think we have WI.Throttler that should make it easier to set this interval and address some of the issues I mentioned, but it might not be async-ready currently